src/Language/Haskell/Liquid/Bare/OfType.hs

```diff
@@ -148,6 +143,24 @@ ofBRType appRTAlias resolveReft
     go_syms
       = secondM ofBSort
 
+    goRFun bounds _ (RApp c ps _ _) t | Just bnd <- M.lookup c bounds 
+      = makeBound bnd (bar <$> ps) <$> go t
+    goRFun _ x t1 t2
+      = rFun x <$> go t1 <*> go t2
+
+    bar (RVar x _) = x
+    bar _          = errorstar "BAR"   
```

Comment:
Right, but we could also have a "top-level" `catch` somewhere to catch the exception from `error`.